### PR TITLE
lookahead collator: only build on top of scheduled relay parents

### LINKED ISSUE #1400: Lookahead Collator should build only on scheduled relay-parents
Currently, the lookahead collator assumes that the chain is scheduled at every block. However, if the lookahead collator builds on a relay parent where the parachain is not scheduled, this candidate can never be included.

The lookahead collator should only build on chains where all in-scope and off-chain blocks are built on allowed relay-parents.

### PR DESCRIPTION
**Pull request by slumber**
Partially addresses #1400

### CODE REVIEW

**Comment by sandreim** [comment]
With async backing enabled, no candidates are backed with this PR and latest Westend(master) runtime.

**Comment by sandreim** [comment]
> With async backing enabled, no candidates are backed with this PR and latest Westend(master) runtime.

Actually I haven't built it with `network-protocol-staging` and that explains the issue.

File: cumulus/client/consensus/aura/src/collators/lookahead.rs [hunk 1]
```diff
@@ -180,6 +182,10 @@ where
 		while let Some(relay_parent_header) = import_notifications.next().await {
 			let relay_parent = relay_parent_header.hash();
 
+			if !is_para_scheduled(relay_parent, params.para_id, &mut params.overseer_handle).await {
+				continue
```
Comment (sandreim) [inline cumulus/client/consensus/aura/src/collators/lookahead.rs:186]: A log trace would be helpful.

**Review by sandreim — APPROVED** [review]

**Comment by sandreim** [comment]
Looks good on Versi!

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3615596

**Comment by rphmeier** [comment]
> The lookahead collator should only build on chains where all in-scope and off-chain blocks are built on allowed relay-parents.

This PR only does half of the issue.
It changes collators not to build on disallowed relay-parents, but collators will still build on top of blocks which had disallowed relay parents. The parent search would need to be altered in order to fix this.

I'm OK with merging as-is, since an entirely-honest collator set would have no issues. But we should leave #1400 open

**Review by rphmeier — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3659191

**Pull request merged**